net/isoping/isoping.go

```diff
@@ -0,0 +1,203 @@
+
+	latRx       int64
+	latRxMin    int64
+	latRxMax    int64
+	latRxCount  int64
+	latRxSum    int64
+	latRxVarSum int64
+}
+
+// Incremental standard deviation calculation, without needing to know the
+// mean in advance.  See:
+// http://mathcentral.uregina.ca/QQ/database/QQ.09.02/carlos1.html
+func onepass_stddev(sumsq int64, sum int64, count int64) float64 {
+	numer := (count * sumsq) - (sum * sum)
+	denom := count * (count - 1)
+	return math.Sqrt(DIV(numer, denom))
+}
+
+// ustimenow subtracts the time since the program started and returns it
+func (srv *Isoping) ustimenow() uint64 {
+	tn := time.Since(srv.ClockStartTime)
+	return uint64(tn.Microseconds())
+}
+
+// Ustime casts the result of ustimenow to uint32 and returns it
+func (srv *Isoping) Ustime() uint32 {
+	return uint32(srv.ustimenow())
+}
+
+// initClock keeps track of when the server/client starts.
+// keeps the exact time and we can subtract from the time
+// to get monotonicClock values
+func (srv *Isoping) initClock() {
+	srv.ClockStartTime = time.Now()
+}
+
+// initClient sets the Isoping.Conn, to the address string otherwise
+// uses [::]:4948 as the default
+func (srv *Isoping) initClient(addressString string) {
+	srv.initClock()
+	srv.IsServer = false
+	udpaddr, err := net.ResolveUDPAddr("udp6", addressString)
+	if err != nil {
+		log.Println(err)
+		addr := "[::]" + SERVER_PORT
+		udpaddr, err = net.ResolveUDPAddr("udp6", addr)
+		if err != nil {
+			log.Println(err)
+			return
+		}
+		log.Printf("Address %v failed to resolve, using %v instead\n", addressString, udpaddr)
+	}
+
+	conn, err := net.DialUDP("udp6", nil, udpaddr)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	srv.RemoteAddr = udpaddr
+	srv.Conn = conn
+}
+
+// initServer sets the Conn field of Isoping, for the listener side.
+func (srv *Isoping) initServer() {
+	srv.initClock()
+	srv.IsServer = true
+	addr, err := net.ResolveUDPAddr("udp6", SERVER_PORT)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	srv.ListenAddr = addr
+	srv.Conn, err = net.ListenUDP("udp", addr)
+	if err != nil {
+		log.Printf("%v\n", err)
+		return
+	}
+}
+
+// initVars initializes a lot of the necessary variables for the calculation
+func (srv *Isoping) initVars() {
+	srv.nextTxId = 1
+	srv.nextRxId = 0
+
+	srv.nextRxackId = 0
+	srv.startRtxtime = 0
+	srv.startRxtime = 0
+	srv.lastRxtime = 0
+
+	srv.minCycleRxdiff = 0
+	srv.nextCycle = 0
+	srv.now = srv.Ustime()
+	srv.nextSend = 0
+	srv.nextTxackIndex = 0
+	srv.Tx = Packet{}
+	srv.Rx = Packet{}
+
+	srv.LastAckInfo = ""
+	srv.lastPrint = srv.now - uint32(srv.usecPerPkt)
+	srv.latTx, srv.latTxMin, srv.latTxMax = 0, 0x7fffffff, 0
+	srv.latTxCount, srv.latTxMin, srv.latTxVarSum = 0, 0, 0
+
+	srv.latRx, srv.latRxMin, srv.latRxMax = 0, 0x7fffffff, 0
+	srv.latRxCount, srv.latRxMin, srv.latRxVarSum = 0, 0, 0
+}
+
+// generateInitialPacket generates the inital packet Tx
+func (srv *Isoping) generateInitialPacket() (*bytes.Buffer, error) {
+	srv.Tx.Magic = MAGIC
```

Comment:
Idiomatic Go would put most of these static initializations into the struct creation as named fields.